src/odbc_connection.cpp

```diff
@@ -1809,8 +1809,15 @@ class CallProcedureAsyncWorker : public ODBCAsyncWorker {
 
       data->deleteColumns(); // delete data in columns for next result set
 
-      data->sql = new SQLTCHAR[255]();
+      data->sql = new SQLTCHAR[255];
+#ifndef UNICODE
       sprintf((char *)data->sql, "{ CALL %s (%s) }", combinedProcedureName, parameterString);
+#else
+      // Note: On Windows, %s and %S change their behavior depending on whether
+      // it's passed to a printf function or a wprintf function. Since we're passing
+      // a narrow string to a wide function, we need to use %S.
+      swprintf(data->sql, 255, L"{ CALL %S (%S) }", combinedProcedureName, parameterString);
```

Comment:
Is this not a wide string passed to a wide function? If `UNICODE` is defined, `data->sql` will be an `SQLWCHAR`, and you create a wide string with `L"..."`, but maybe there is something I don't get.

Reply:
I was referring to the values associated with the %S, ie. combinedProcedureName and parameterString, though I see that parameterString actually has the same problem where it's SQLTCHAR but being created as a C string. I'll fix that too.